integration-tests/test/test_backward_compatible.py

```diff
@@ -0,0 +1,84 @@
+
+CEREMONY_MASTER_PRIVATE = PrivateKey.from_hex("80366db5fbb8dad7946f27037422715e4176dda41d582224db87b6c3b783d709")
+VALIDATOR_A_PRIVATE = PrivateKey.from_hex("120d42175739387af0264921bb117e4c4c05fbe2ce5410031e8b158c6e414bb5")
+VALIDATOR_B_PRIVATE = PrivateKey.from_hex("1f52d0bce0a92f5c79f2a88aae6d391ddf853e2eb8e688c5aa68002205f92dad")
+
+
+@contextlib.contextmanager
+def temp_rnode_data(mount_dir: str) -> Generator[str, None, None]:
+    with tempfile.TemporaryDirectory() as temp_dir:
+        rnode_data_dir = get_absolute_path_for_mounting("rnode_data", mount_dir)
+        with tarfile.open(os.path.join(rnode_data_dir, 'bootstrap')) as tar:
+            tar.extractall(os.path.join(temp_dir, 'bootstrap'))
+        with tarfile.open(os.path.join(rnode_data_dir, 'validatorA')) as tar:
+            tar.extractall(os.path.join(temp_dir, 'validatorA'))
+        with tarfile.open(os.path.join(rnode_data_dir, 'validatorB')) as tar:
+            tar.extractall(os.path.join(temp_dir, 'validatorB'))
+        yield temp_dir
+
+def test_backward_compatible(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    """
+    This test is for backward compatibility. It use the old version data of a rnode the create a existing network.
+    If your pr failed on this test, it means your update make the old version rnode not working normally.
+
+    Currently we don't release our mainnet so it is Ok to fail the test. Once your pr make the test failed , you
+    should shout out loudly in discord and let everybody know your update is not backward compatible.
+    """
+    with  testing_context(command_line_options, random_generator, docker_client) as context, \
+        temp_rnode_data(context.mount_dir) as temp_rnode, \
+        docker_network(context, context.docker) as network, \
+        started_bootstrap(context=context, network=network, extra_volumes=["{}/bootstrap/rnode:/var/lib/rnode".format(temp_rnode)]) as bootstrap_node, \
+        started_peer(context=context, network=network, bootstrap=bootstrap_node, name='validator-a', private_key=VALIDATOR_A_PRIVATE, extra_volumes=["{}/validatorA/rnode:/var/lib/rnode".format(temp_rnode)], synchrony_constraint_threshold=0.33) as validator_a, \
+        started_peer(context=context, network=network, bootstrap=bootstrap_node, name='validator-b', private_key=VALIDATOR_B_PRIVATE, extra_volumes=["{}/validatorB/rnode:/var/lib/rnode".format(temp_rnode)], synchrony_constraint_threshold=0.33) as validator_b:
+
+        wait_for_approved_block_received_handler_state(context, bootstrap_node)
+        wait_for_approved_block_received_handler_state(context, validator_a)
+        wait_for_approved_block_received_handler_state(context, validator_b)
+        wait_for_peers_count_at_least(context, bootstrap_node, 2)
+        wait_for_peers_count_at_least(context, validator_a, 2)
+        wait_for_peers_count_at_least(context, validator_b, 2)
+
+
+        # why is here 28?
```

Comment:
should be 22 here?

Reply:
You are right. I forgot to update the comments.